### ISSUE TITLE
Option to download in compressed format does not work

### ISSUE THREAD

**Issue by damonLL** [quasar-analytics/quasar#1858]
This is related to SlamData Issue # [1370](https://github.com/slamdata/slamdata/issues/1370)

The ability to download a SlamData / Quasar File (not a SlamData workspace) in compressed format does not appear to work from the SlamData UI.

Hopefully linking the two issues is helpful, and not just overhead :)

It's unclear whether this is an issue with frontend, backend or both.

**Comment by sellout** [comment]
Well, on the plus side, Quasar already knows how to do the compression – we just need to support a way for the client to request the result in compressed format. I _think_ the best way to do this is to have the client include `Accept: application/json+gzip` in the request. (Not that that’s supported by Quasar yet).

Also, I would say that the existing `Accept-Encoding: gzip` should be included in most queries, etc., unless 1. we know the response is small (e.g., getting counts for pagination) or 2. the user requested compressed data (in which case you want to use the above `Accept` rather than `Accept-Encoding`, since we don’t want to try to gzip the gzipped content).

@natefaubion Does that sound reasonable?

**Comment by natefaubion** [comment]
@sellout I think that should be fine.

**Comment by jdegoes** [comment]
@zeeiyerWork Can you make sure this gets done in 4.2 for the front-end team? We need it for workspace import / export.

**Comment by zeeiyerWork** [comment]
@natefaubion -> Can we fix this in 4.2 or is this already resolved via the issues/PR you reference? 
thanks

**Comment by zeeiyerWork** [comment]
@sellout - the front-end is still blocked by this - so I have added it to the May iteration.

**Comment by sellout** [comment]
@nic has graciously agreed to give this one a go.

**Comment by nic** [comment]
@sellout Ok! 👍

**Comment by nicflores** [comment]
@sellout, I'm @nicflores on GitHub, @nic was already taken.

**Comment by nicflores** [comment]
@natefaubion, I didn't want this feature to slip through the cracks, but maybe its already on your radar or someone else on the fronted team.

Here are a few more details. Upon file download the UI uses a get request of the form:
```
GET http://localhost:8080/data/fs//mongo32/local/smallZips?request-headers=%7B%22Accept-Encoding%22%3A%22gzip%22%2C%22Accept%22%3A%22application%2Fjson%3Bdisposition%3D%5C%22attachment%3B%20filename*%3DUTF-8''smallZips.json%5C%22%22%7D
← 200 OK application/json 2.02k 323ms
```
Within the url query parameter we have the following headers:
```
request-headers: {"Accept-Encoding":"gzip","Accept":"application/json;disposition=\"attachment; filename*=UTF-8''smallZips.json\""}
```

The new functionality added is as follows. If you modify the headers in the query parameter as follows:
```
request-headers: {"Accept-Encoding":"gzip","Accept":"application/zip,application/json;disposition=\"attachment; filename*=UTF-8''smallZips.json.zip\""}
```
you will get a zipped file called `smallZips.json.zip`.

I hope this helps move along the UI feature requested by @damonLL.

**Comment by garyb** [comment]
@nicflores thanks, zip download seems to be working great with that!

**Comment by damonLL** [comment]
I tried this, but maybe I'm misunderstanding the status?

I'm able to now download a file with the extension of `.zip` but it still looks like it's in normal text format. Meaning I can `cat filename.zip` and read it as plain JSON or CSV.

Also, if I select `CSV` and zipped, I get a file called `filename`. If I select `JSON` and zipped, I get a file called `filename.zip`.  There is no `.csv.zip` or `.json.zip` in the filename.

@nicflores and @garyb - is this expected behavior? Is there another unimplemented Issue I'm missing?  Thanks!

**Comment by garyb** [comment]
@damonLL it's only just been merged into master on the front end, but it definitely works there with the latest Quasar!
